todo_app/app.py

```diff
@@ -1,15 +1,60 @@
 from flask import Flask, render_template, request, redirect, url_for, Blueprint
 from todo_app.data import trello_items as items
 from todo_app.view_model import ViewModel
-from todo_app.flask_config import Config
+from todo_app.flask_config import Config, get_access_token, get_user_data
+from flask_login import LoginManager, login_required, UserMixin, login_user, current_user
+
+class User(UserMixin):
+    def __init__(self, id):
+        self.id = id
+    
+    @property
+    def role(self):
+        print(id)
+        if self.id == '94120411':
+            return "writer"
+        else:
+            return "reader"
+
 
 def create_app():
 
     app = Flask(__name__)
     app.config.from_object(Config())
 
+    login_manager = LoginManager()
+
+    @login_manager.unauthorized_handler
+    def unauthenticated():
+        return redirect('https://github.com/login/oauth/authorize?client_id=' + app.config['CLIENT_ID'] + '&state=' + app.config['STATE'])
```

Comment:
This suggests that state probably isn't being used very helpfully here - having it set once per app makes it quite easy for someone to just find out (since it's in their browser history) and then it doesn't really achieve anything. If we're using it for security (as opposed to it's potential use as state like telling us which page they were trying to visit before being redirect) then the ideal flow is:
* Set the state to a random value for this user
* Store the same random value on their session cookie
* When they get redirected with a `code`, also check that the state returned matches the state in their session cookie, and only allow them through if it does